process/block/metablock.go

```diff
@@ -262,7 +263,7 @@ func (mp *metaProcessor) removeBlockInfoFromPool(header *block.MetaBlock) error
 		}
 
 		headerPool.Remove(shardData.HeaderHash)
-		headerNoncesPool.Remove(hdr.Nonce)
+		headerNoncesPool.RemoveNonce(hdr.Nonce)
```

Comment:
this remove is not good here. As not all the shardHeaders with hdr.Nonce are finished. This code contained bug before. Remove should be for nonces < min(lastNotarizedHdrs)

Reply:
changed to `headerNoncesPool.RemoveShardId(hdr.Nonce, hdr.ShardId)`